aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/jaxrs/AwsProxySecurityContext.java

```diff
@@ -83,12 +83,7 @@ public Principal getUserPrincipal() {
         if (getAuthenticationScheme().equals(AUTH_SCHEME_CUSTOM) || getAuthenticationScheme().equals(AUTH_SCHEME_AWS_IAM)) {
             return () -> {
                 if (getAuthenticationScheme().equals(AUTH_SCHEME_CUSTOM)) {
-                    switch (event.getRequestSource()) {
-                    case API_GATEWAY:
-                        return event.getRequestContext().getAuthorizer().getPrincipalId();
-                    case ALB:
-                        return event.getMultiValueHeaders().getFirst(ALB_IDENTITY_HEADER);
-                    }
+                    return event.getRequestContext().getAuthorizer().get("principalId").toString();
```

Comment:
Please define a constant for `principalId` (ideally it should be provided by aws-lambda-events lib).

Reply:
Updated